File: solutions/de.vimotest.sandbox/source_gen/de/vimotest/sandbox/commands/SelectRowCommand_ViewModel.cpp

#include "SelectRowCommand_ViewModel.h"
#include <vector>
#include <memory>
#include "SelectRowCommand_ViewModelMyListViewElementsRow.h"
#include "SelectRowCommand_ViewModelMyStringRowHandleListViewElementsRow.h"
#include "SelectRowCommand_ViewModelMyTableViewElementsRow.h"
#include "SelectRowCommand_ViewModelMyStringRowHandleTableViewElementsRow.h"
#include "SelectRowCommand_ViewModelMyTreeViewElementsRow.h"
#include "SelectRowCommand_ViewModelMyStringRowHandleTreeViewElementsRow.h"
#include "SelectRowCommand_ViewModelMyIndexBasedMultiSelectRowListViewElementsRow.h"
#include "SelectRowCommand_ViewModelMyStringHandleMultiSelectRowListViewElementsRow.h"
#include <string>

namespace commands
{
  std::vector<std::shared_ptr<SelectRowCommand_ViewModelMyListViewElementsRow>> SelectRowCommand_ViewModel::getMyListViewElementsListRows( ) 
  {
    return this->MyListViewElementsListRows;
  }
  int SelectRowCommand_ViewModel::getMyListViewElementsListSelectedRow( ) 
  {
    return this->MyListViewElementsListSelectedRow;
  }
  std::vector<std::shared_ptr<SelectRowCommand_ViewModelMyStringRowHandleListViewElementsRow>> SelectRowCommand_ViewModel::getMyStringRowHandleListViewElementsListRows( ) 
  {
    return this->MyStringRowHandleListViewElementsListRows;
  }
  std::string SelectRowCommand_ViewModel::getMyStringRowHandleListViewElementsListSelectedRow( ) 
  {
    return this->MyStringRowHandleListViewElementsListSelectedRow;
  }
  std::vector<std::shared_ptr<SelectRowCommand_ViewModelMyTableViewElementsRow>> SelectRowCommand_ViewModel::getMyTableViewElementsTableRows( ) 
  {
    return this->MyTableViewElementsTableRows;
  }
  int SelectRowCommand_ViewModel::getMyTableViewElementsTableSelectedRow( ) 
  {
    return this->MyTableViewElementsTableSelectedRow;
  }
  std::vector<std::shared_ptr<SelectRowCommand_ViewModelMyStringRowHandleTableViewElementsRow>> SelectRowCommand_ViewModel::getMyStringRowHandleTableViewElementsTableRows( ) 
  {
    return this->MyStringRowHandleTableViewElementsTableRows;
  }
  std::string SelectRowCommand_ViewModel::getMyStringRowHandleTableViewElementsTableSelectedRow( ) 
  {
    return this->MyStringRowHandleTableViewElementsTableSelectedRow;
  }
  std::vector<std::shared_ptr<SelectRowCommand_ViewModelMyTreeViewElementsRow>> SelectRowCommand_ViewModel::getMyTreeViewElementsTreeRows( ) 
  {
    return this->MyTreeViewElementsTreeRows;
  }
  int SelectRowCommand_ViewModel::getMyTreeViewElementsTreeSelectedRow( ) 
  {
    return this->MyTreeViewElementsTreeSelectedRow;
  }
  std::vector<std::shared_ptr<SelectRowCommand_ViewModelMyStringRowHandleTreeViewElementsRow>> SelectRowCommand_ViewModel::getMyStringRowHandleTreeViewElementsTreeRows( ) 
  {
    return this->MyStringRowHandleTreeViewElementsTreeRows;
  }
  std::string SelectRowCommand_ViewModel::getMyStringRowHandleTreeViewElementsTreeSelectedRow( ) 
  {
    return this->MyStringRowHandleTreeViewElementsTreeSelectedRow;
  }
  std::vector<std::shared_ptr<SelectRowCommand_ViewModelMyIndexBasedMultiSelectRowListViewElementsRow>> SelectRowCommand_ViewModel::getMyIndexBasedMultiSelectRowListViewElementsListRows( ) 
  {
    return this->MyIndexBasedMultiSelectRowListViewElementsListRows;
  }
  std::vector<int> SelectRowCommand_ViewModel::getMyIndexBasedMultiSelectRowListViewElementsListSelectedRows( ) 
  {
    return this->MyIndexBasedMultiSelectRowListViewElementsListSelectedRows;
  }
  std::vector<std::shared_ptr<SelectRowCommand_ViewModelMyStringHandleMultiSelectRowListViewElementsRow>> SelectRowCommand_ViewModel::getMyStringHandleMultiSelectRowListViewElementsListRows( ) 
  {
    return this->MyStringHandleMultiSelectRowListViewElementsListRows;
  }
  std::vector<std::string> SelectRowCommand_ViewModel::getMyStringHandleMultiSelectRowListViewElementsListSelectedRows( ) 
  {
    return this->MyStringHandleMultiSelectRowListViewElementsListSelectedRows;
  }
  void SelectRowCommand_ViewModel::myListViewElementsRowSelected(int rowIndex ) 
  {
    
  }
  void SelectRowCommand_ViewModel::myStringRowHandleListViewElementsRowSelected(std::string rowHandle ) 
  {
    
  }
  void SelectRowCommand_ViewModel::myTableViewElementsRowSelected(int rowIndex ) 
  {
    
  }
  void SelectRowCommand_ViewModel::myStringRowHandleTableViewElementsRowSelected(std::string rowHandle ) 
  {
    
  }
  void SelectRowCommand_ViewModel::myTreeViewElementsRowSelected(int rowIndex ) 
  {
    
  }
  void SelectRowCommand_ViewModel::myStringRowHandleTreeViewElementsRowSelected(std::string rowHandle ) 
  {
    
  }
  void SelectRowCommand_ViewModel::myIndexBasedMultiSelectRowListViewElementsRowSelected(std::vector<int> rowIndices ) 
  {
    
  }
  void SelectRowCommand_ViewModel::myStringHandleMultiSelectRowListViewElementsRowSelected(std::vector<std::string> rowHandles ) 
  {
    
  }
}

File: solutions/de.vimotest.sandbox/source_gen/de/vimotest/sandbox/commands/SelectRowCommand_ViewModel.cs
/// <filename>
///     SelectRowCommand_ViewModel.cs
/// </filename>
using System.Collections.Generic;
using commands;

namespace commands
{
    public class SelectRowCommand_ViewModel
    {
        private System.Collections.Generic.List<commands.SelectRowCommand_ViewModelMyListViewElementsRow> MyListViewElementsListRows = new System.Collections.Generic.List<commands.SelectRowCommand_ViewModelMyListViewElementsRow>();

        public virtual System.Collections.Generic.List<commands.SelectRowCommand_ViewModelMyListViewElementsRow> getMyListViewElementsListRows()
        {
            return this.MyListViewElementsListRows;
        }

        private int MyListViewElementsListSelectedRow;

        public virtual int getMyListViewElementsListSelectedRow()
        {
            return this.MyListViewElementsListSelectedRow;
        }

        private System.Collections.Generic.List<commands.SelectRowCommand_ViewModelMyStringRowHandleListViewElementsRow> MyStringRowHandleListViewElementsListRows = new System.Collections.Generic.List<commands.SelectRowCommand_ViewModelMyStringRowHandleListViewElementsRow>();

        public virtual System.Collections.Generic.List<commands.SelectRowCommand_ViewModelMyStringRowHandleListViewElementsRow> getMyStringRowHandleListViewElementsListRows()
        {
            return this.MyStringRowHandleListViewElementsListRows;
        }

        private string MyStringRowHandleListViewElementsListSelectedRow;

        public virtual string getMyStringRowHandleListViewElementsListSelectedRow()
        {
            return this.MyStringRowHandleListViewElementsListSelectedRow;
        }

        private System.Collections.Generic.List<commands.SelectRowCommand_ViewModelMyTableViewElementsRow> MyTableViewElementsTableRows = new System.Collections.Generic.List<commands.SelectRowCommand_ViewModelMyTableViewElementsRow>();

        public virtual System.Collections.Generic.List<commands.SelectRowCommand_ViewModelMyTableViewElementsRow> getMyTableViewElementsTableRows()
        {
            return this.MyTableViewElementsTableRows;
        }

        private int MyTableViewElementsTableSelectedRow;

        public virtual int getMyTableViewElementsTableSelectedRow()
        {
            return this.MyTableViewElementsTableSelectedRow;
        }

        private System.Collections.Generic.List<commands.SelectRowCommand_ViewModelMyStringRowHandleTableViewElementsRow> MyStringRowHandleTableViewElementsTableRows = new System.Collections.Generic.List<commands.SelectRowCommand_ViewModelMyStringRowHandleTableViewElementsRow>();

        public virtual System.Collections.Generic.List<commands.SelectRowCommand_ViewModelMyStringRowHandleTableViewElementsRow> getMyStringRowHandleTableViewElementsTableRows()
        {
            return this.MyStringRowHandleTableViewElementsTableRows;
        }

        private string MyStringRowHandleTableViewElementsTableSelectedRow;

        public virtual string getMyStringRowHandleTableViewElementsTableSelectedRow()
        {
            return this.MyStringRowHandleTableViewElementsTableSelectedRow;
        }

        private System.Collections.Generic.List<commands.SelectRowCommand_ViewModelMyTreeViewElementsRow> MyTreeViewElementsTreeRows = new System.Collections.Generic.List<commands.SelectRowCommand_ViewModelMyTreeViewElementsRow>();

        public virtual System.Collections.Generic.List<commands.SelectRowCommand_ViewModelMyTreeViewElementsRow> getMyTreeViewElementsTreeRows()
        {
            return this.MyTreeViewElementsTreeRows;
        }

        private int MyTreeViewElementsTreeSelectedRow;

        public virtual int getMyTreeViewElementsTreeSelectedRow()
        {
            return this.MyTreeViewElementsTreeSelectedRow;
        }

        private System.Collections.Generic.List<commands.SelectRowCommand_ViewModelMyStringRowHandleTreeViewElementsRow> MyStringRowHandleTreeViewElementsTreeRows = new System.Collections.Generic.List<commands.SelectRowCommand_ViewModelMyStringRowHandleTreeViewElementsRow>();

        public virtual System.Collections.Generic.List<commands.SelectRowCommand_ViewModelMyStringRowHandleTreeViewElementsRow> getMyStringRowHandleTreeViewElementsTreeRows()
        {
            return this.MyStringRowHandleTreeViewElementsTreeRows;
        }

        private string MyStringRowHandleTreeViewElementsTreeSelectedRow;

        public virtual string getMyStringRowHandleTreeViewElementsTreeSelectedRow()
        {
            return this.MyStringRowHandleTreeViewElementsTreeSelectedRow;
        }

        private System.Collections.Generic.List<commands.SelectRowCommand_ViewModelMyIndexBasedMultiSelectRowListViewElementsRow> MyIndexBasedMultiSelectRowListViewElementsListRows = new System.Collections.Generic.List<commands.SelectRowCommand_ViewModelMyIndexBasedMultiSelectRowListViewElementsRow>();

        public virtual System.Collections.Generic.List<commands.SelectRowCommand_ViewModelMyIndexBasedMultiSelectRowListViewElementsRow> getMyIndexBasedMultiSelectRowListViewElementsListRows()
        {
            return this.MyIndexBasedMultiSelectRowListViewElementsListRows;
        }

        private System.Collections.Generic.List<int> MyIndexBasedMultiSelectRowListViewElementsListSelectedRows = new System.Collections.Generic.List<int>();

        public virtual System.Collections.Generic.List<int> getMyIndexBasedMultiSelectRowListViewElementsListSelectedRows()
        {
            return this.MyIndexBasedMultiSelectRowListViewElementsListSelectedRows;
        }

        private System.Collections.Generic.List<commands.SelectRowCommand_ViewModelMyStringHandleMultiSelectRowListViewElementsRow> MyStringHandleMultiSelectRowListViewElementsListRows = new System.Collections.Generic.List<commands.SelectRowCommand_ViewModelMyStringHandleMultiSelectRowListViewElementsRow>();

        public virtual System.Collections.Generic.List<commands.SelectRowCommand_ViewModelMyStringHandleMultiSelectRowListViewElementsRow> getMyStringHandleMultiSelectRowListViewElementsListRows()
        {
            return this.MyStringHandleMultiSelectRowListViewElementsListRows;
        }

        private System.Collections.Generic.List<string> MyStringHandleMultiSelectRowListViewElementsListSelectedRows = new System.Collections.Generic.List<string>();

        public virtual System.Collections.Generic.List<string> getMyStringHandleMultiSelectRowListViewElementsListSelectedRows()
        {
            return this.MyStringHandleMultiSelectRowListViewElementsListSelectedRows;
        }

        public virtual void myListViewElementsListRowSelected(int rowIndex)
        {
        }

        public virtual void myStringRowHandleListViewElementsListRowSelected(string rowHandle)
        {
        }

        public virtual void myTableViewElementsTableRowSelected(int rowIndex)
        {
        }

        public virtual void myStringRowHandleTableViewElementsTableRowSelected(string rowHandle)
        {
        }

        public virtual void myTreeViewElementsTreeRowSelected(int rowIndex)
        {
        }

        public virtual void myStringRowHandleTreeViewElementsTreeRowSelected(string rowHandle)
        {
        }

        public virtual void myIndexBasedMultiSelectRowListViewElementsListRowSelected(System.Collections.Generic.List<int> rowIndices)
        {
        }

        public virtual void myStringHandleMultiSelectRowListViewElementsListRowSelected(System.Collections.Generic.List<string> rowHandles)
        {
        }
    }
}
